internal/domain/domain_operations.go

```diff
@@ -0,0 +1,242 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package domain
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/terraform-provider-vcf/internal/cluster"
+	"github.com/vmware/terraform-provider-vcf/internal/constants"
+	"github.com/vmware/terraform-provider-vcf/internal/network"
+	validationUtils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/terraform-provider-vcf/internal/vcenter"
+	"github.com/vmware/vcf-sdk-go/client"
+	"github.com/vmware/vcf-sdk-go/client/clusters"
+	"github.com/vmware/vcf-sdk-go/client/domains"
+	"github.com/vmware/vcf-sdk-go/models"
+	"sort"
+)
+
+func CreateDomainCreationSpec(data *schema.ResourceData) (*models.DomainCreationSpec, error) {
+	result := new(models.DomainCreationSpec)
+	domainName := data.Get("name").(string)
+	result.DomainName = &domainName
+
+	if orgName, ok := data.GetOk("org_name"); ok {
+		result.OrgName = orgName.(string)
+	}
+
+	vcenterSpec, err := generateVcenterSpecFromResourceData(data)
+	if err == nil {
+		result.VcenterSpec = vcenterSpec
+	} else {
+		return nil, err
+	}
+
+	nsxSpec, err := generateNsxSpecFromResourceData(data)
```

Comment:
nsxTSpec?

Reply:
Let's keep it as nsxSpec. NSX-T has been rebranded to NSX recently.